contracts/most/UtBase.sol

```diff
@@ -0,0 +1,260 @@
+         * genesisTotalSupply.
+         */
+        balances[address(this)] = genesisTotalSupply;
+    }
+
+    /**
+     * @notice Mints the ERC20 token for the beneficiary address and
+     *         unwraps its to base coin.
+     *
+     * \pre Caller must be Consensus cogateway contract.
+     *
+     * \post Increases total token supply.
+     * \post Increase ERC20 UtBase token balance of the beneficiary.
+     * \post It satisfies \post conditions of UtBase::unwrapInternal().
+     *
+     * @param _beneficiary Beneficairy address which will receive base coin.
+     * @param _amount Amount of UtBase tokens to be minted.
+     *
+     */
+    function mint(
+        address payable _beneficiary,
+        uint256 _amount
+    )
+        external
+        onlyConsensusCogateway
+    {
+        _mint(_beneficiary, _amount);
+        unwrapInternal(_beneficiary, _amount);
+    }
+
+    /**
+     * @notice Unwrap converts ERC20 UtBase to base coin.
+     *
+     * @dev This contract's base coin balance must always be greater than
+     *       amount.
+     *
+     * \pre Satisfies \pre conditions of UtBase::unwrapInternal.
+     *
+     * \post Satisfies \post conditions of UtBase::unwrapInternal.
+     *
+     * @param _amount Amount of ERC20 UtBase token to convert to base coin.
+     */
+    function unwrap(
+        uint256 _amount
+    )
+        external
+    {
+        unwrapInternal(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Wrap converts base coin to ERC20 UtBase.
+     *
+     * @dev The ERC20 OST balance of contract should always be greater than the
+     *      received payable amount.
+     *
+     * \pre msg.value should be non-zero.
+     * \pre UtBase ERC20 token balance must be greater than
+     *      amount to be wrapped.
+     *
+     * \post Increases ERC20 UtBase token balance of the caller.
+     * \post Decreases ERC20 UtBase token balance of the UtBase contract.
+     */
+    function wrap()
+        external
+        payable
+    {
+        uint256 amount = msg.value;
+        address account = msg.sender;
+
+        require(
+            amount > 0,
+            "Payable amount should not be zero."
+        );
+
+        assert(balances[address(this)] >= amount);
+
+        transferBalance(address(this), account, amount);
+    }
+
+    /**
+     * @notice It allows to burn UtBase tokens for an account.
+     *
+     * \pre msg.value should be non-zero.
+     * \pre Caller must have atleast `_value` amount of ERC20 UtBase tokens.
+     *
+     * \post Decreases total token supply of the token.
+     * \post Decreases caller's ERC20 UtBase token balance.
+     *
+     * @param _value The amount that will be burnt.
+     */
+    function burn(uint256 _value)
+        external
+    {
+        _burn(msg.sender, _value);
+    }
+
+    /**
+     * @notice It allows to burn tokens of the spender.
+     *
+     * \pre `_account` must approve `_value` amount to the caller
```

Comment:
Please place this after `@param` section

Reply:
Done